AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m

```diff
@@ -250,13 +250,17 @@ - (MSDevice *)deviceForTimestamp:(NSDate *)timestamp {
 
 - (void)clearDevices {
   @synchronized(self) {
+      
+    // Refresh current device info if needed before archive.
+    [self device];
```

Comment:
Why do you refresh the current device info? I think it is already there when device tracker was started.

Reply:
No, only if the history of the devices was empty, the current device will be added to the history (based on the constructor). Therefore, we need to make sure that the current device is really added to the history.

Reply:
> only if the history of the devices was empty

I think your assumption is wrong here. It will add device info whenever device tracker is started. Can you see the issue without this line?

Reply:
I fixed the solution, now when initializing the tracker, we always save the current device.